### PR TITLE
Modal checkout – post-checkout newsletter signup

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds post-checkout newsletter signup options:

image

image

### How to test the changes in this Pull Request:

1. Switch `newspack-plugin` to branch from https://github.com/Automattic/newspack-plugin/pull/2707
1. Switch `newspack-theme` to branch from https://github.com/Automattic/newspack-theme/pull/2188 
1. Ensure that the Newsletters provider is configured and some lists are active in Engagement wizard -> Newsletters
1. With "Newspack" set as the Reader Revenue platform, make a donation using the checkout modal
3. Observe that after the checkout is successful, the chosen newsletter lists are presented
4. Select some of them and click "Continue", observe a success message
5. Verify that the email used has been added to the list(s)
6. Set a `NEWSPACK_DISABLE_POST_CHECKOUT_NEWSLETTER_SIGNUP` environment variable to `true` and observe that now the signup lists are not displayed after the checkout 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/modal-checkout/modal.js [hunk 1]
```diff
@@ -83,17 +89,15 @@ domReady( () => {
...
-				button.addEventListener( 'click', ev => {
-					ev.preventDefault();
-					closeCheckout( variationModal );
-				} );
+		variationModal.querySelectorAll( '.newspack-blocks-modal__close' ).forEach( button => {
+			button.addEventListener( 'click', ev => {
+				ev.preventDefault();
+				closeCheckout();
 			} );
+		} );
```
Comment (miguelpeixe) [inline src/modal-checkout/modal.js:100]: This change breaks the CSS for the variation selection modal:

| Before | After |
| --- | --- |
| image |  image |

**Review by miguelpeixe — CHANGES_REQUESTED** [review]

File: src/modal-checkout/modal.js (hunk 1 above)
Comment (adekbadek) [reply src/modal-checkout/modal.js:100]: Thanks for catching that! Fixed in aa52d8f

File: includes/class-modal-checkout.php [hunk 2]
```diff
@@ -589,5 +590,166 @@ class="button"
...
+			</div>
+		<?php
+	}
+
+	/**
+	 * Should post-chcekout newsletter signup be available?
+	 */
+	private static function is_newsletter_signup_available() {
+		return ! ( defined( 'NEWSPACK_DISABLE_POST_CHECKOUT_NEWSLETTER_SIGNUP' ) && NEWSPACK_DISABLE_POST_CHECKOUT_NEWSLETTER_SIGNUP );
+	}
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply includes/class-modal-checkout.php]: Done in e19098fa4ff383bcaa7e63091885349440706b58

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
Thank you for the changes!

It needs rebasing and switch the source of the lists from `Newspack_Newsletters_Subscription::get_lists()` to `Newspack\Reader_Activation::get_registration_newsletter_lists()`. Note that the new source includes a `checked` property that should be used for setting the initial checkbox state.

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
After donating and selecting the lists to subscribe to, I'm redirected to a blank page and the subscription is not processed:

https://github.com/Automattic/newspack-blocks/assets/820752/1dcc43ea-21ab-4d80-8fda-f76652e5d82a

This is due to the `thankyou.php` template bailing on the `$is_valid` check because the `$order` is not passed on the subscription form submission.

**Comment by adekbadek** [comment]
Whoops, good catch! Fixed in 22dc58fa4afb1a79bcd2a0501d30fa3ad393a06f

**Review by miguelpeixe — APPROVED** [review]
Thank you for revising! 🎉

**Pull request merged**